src/user.cpp

```diff
@@ -99,11 +126,23 @@ User* User::unserialize(MegaClient* client, string* d)
     l = *ptr++;
     if (l)
     {
+        if (ptr + l > end)
+        {
+            return NULL;
```

Comment:
The code calling `unserialize()` already checks the resulting value and, if `NULL`, it does log an error.